src/Readers/ODataReader.v4/OdcmReader.cs

```diff
@@ -117,6 +118,16 @@ private void AddVocabularyAnnotations(OdcmObject odcmObject, IEdmVocabularyAnnot
                 odcmObject.Annotations = ODataVocabularyReader.GetOdcmAnnotations(_edmModel, annotatableEdmEntity).ToList();
                 odcmObject.Description = _edmModel.GetDescriptionAnnotation(annotatableEdmEntity);
                 odcmObject.LongDescription = _edmModel.GetLongDescriptionAnnotation(annotatableEdmEntity);
+
+                if(annotatableEdmEntity is IEdmEntitySet && odcmObject is OdcmProperty)
+                {
+                    OdcmCapability.SetCapabilitiesForEntitySet((OdcmProperty)odcmObject);

```

Comment:
I would love to see this get collapsed into the construction of the OdcmProperty, if possible.